dali/pipeline/operators/operator.h

```diff
@@ -213,6 +233,100 @@ class Operator : public OperatorBase {
   template <typename B = Backend>
   typename std::enable_if<!std::is_same<B, GPUBackend>::value>::type
   SyncHelper(int /*unused*/, Workspace<B> */*unused*/) {}
+
+  template <typename B = Backend>
+  typename std::enable_if<std::is_same<B, GPUBackend>::value>::type
+  Flatten(Workspace<Backend> *ws) {
+    seq_sizes_.clear();
+    seq_sizes_.resize(input_sets_, 0);
+    for (int i = 0; i < input_sets_; ++i) {
+      CUDA_CALL(cudaStreamSynchronize(ws->stream()));
+      auto &input = ws->template MutableInput<Backend>(i);
+
+      const std::vector<Dims>& old_shapes = input.shape();
+      DALITensorLayout layout = input.GetLayout();
+      if (layout == DALI_NFHWC || layout == DALI_NFCHW) {
```

Comment:
maybe we should have a `IsSequenceLayout(...)` somewhere

Reply:
Done